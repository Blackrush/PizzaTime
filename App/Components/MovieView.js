import React, { Component } from 'react'
import {
  Dimensions,
  StyleSheet,
  View,
  Text,
  Image,
} from 'react-native'

import ParallaxScrollView from 'react-native-parallax-scroll-view'
import StarRating from '../Components/StarRating'


const PARALLAX_HEIGHT = 450


class MovieView extends Component {
  static propTypes = {
    overview: React.PropTypes.string,
    rating: React.PropTypes.number,
    style: React.PropTypes.any,
    images: React.PropTypes.shape({
      poster: React.PropTypes.shape({
        full: React.PropTypes.string,
      }),
    }),
  }

  _renderPoster = () => {
    const window = Dimensions.get('window')

    return (
      <Image source={{uri: this.props.images.poster.full,
        width: window.width, height: PARALLAX_HEIGHT}}/>
    )
  }

  render() {
    return (
      <ParallaxScrollView parallaxHeaderHeight={PARALLAX_HEIGHT}
                          renderBackground={this._renderPoster}
                          style={this.props.style}>
        <Text style={styles.overview}>
          {this.props.overview}
        </Text>

        <View style={styles.rating}>
          <StarRating size="large" rating={this.props.rating/10*5} maxStars={5}/>
        </View>
      </ParallaxScrollView>
    )
  }
}


const styles = StyleSheet.create({
  overview: {
    padding: 10,
    textAlign: 'justify',
  },
  rating: {
    alignSelf: 'center',
  },
})


export default MovieView