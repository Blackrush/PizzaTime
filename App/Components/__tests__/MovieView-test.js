import React from 'react'
import 'react-native'

import MovieView from '../MovieView'
import { create as render } from 'react-test-renderer'

jest.mock('react-native-parallax-scroll-view', () => 'ParallaxScrollView');

test('Movies renders correctly', () => {
  const movie = require('./movie.json')
  const tree = render(
    <MovieView {...movie}/>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
