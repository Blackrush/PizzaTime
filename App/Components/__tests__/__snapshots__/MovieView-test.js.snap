exports[`test Movies renders correctly 1`] = `
<ParallaxScrollView
  parallaxHeaderHeight={450}
  renderBackground={[Function anonymous]}
  style={undefined}>
  <Text
    style={
      Object {
        "padding": 10,
        "textAlign": "justify"
      }
    }>
    A private eye investigates the apparent suicide of a fading porn star in 1970s Los Angeles and uncovers a conspiracy.
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center"
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between"
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F9D53B",
                "fontFamily": "FontAwesome",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "opacity": 1
          }
        }
        testID={undefined}>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gray",
                "fontFamily": "FontAwesome",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal"
              },
              undefined
            ]
          }>
          
        </Text>
      </View>
    </View>
  </View>
</ParallaxScrollView>
`;
